### ISSUE TITLE
Error message "document MID is not unique" missing clear indication of offending requirement and document

### ISSUE THREAD

**Issue by bernhard-tuvsud** [strictdoc-project/strictdoc#1865]
I got the following error message:

```
error: TraceabilityIndex: the document MID is not unique: 1de8e492b37445f79880996ac14b9e5d. All machine identifiers (MID) must be unique values.
```

It is nearly impossible to find the offending requirement or document by this error message.

Please add document and requirement id to error message. This would also beneficial for other error messages.

Full log:

```
Parallelization: Enabled
Step 'Parse SDoc project tree' start.
Step 'Find and read SDoc files' start.
Reading SDOC: isa_assessment.sdoc ........................... 0.02s
Reading SDOC: data_passing.sdoc ............................. 0.00s
Reading SDOC: device_driver_api.sdoc ........................ 0.00s
Reading SDOC: SC_FunctionalSafety_Obl_of_Use.sdoc ........... 0.00s
Reading SDOC: exception_and_error_handling.sdoc ............. 0.00s
Reading SDOC: isa_assessment.sdoc ........................... 0.02s
Reading SDOC: SC_ReqMangement_Checklist.sdoc ................ 0.01s
Reading SDOC: file_system.sdoc .............................. 0.00s
Reading SDOC: SC_SafetyClaims.sdoc .......................... 0.01s
Reading SDOC: hw_arch_interface.sdoc ........................ 0.00s
Reading SDOC: SC_ReqMangement_Strategy.sdoc ................. 0.00s
Reading SDOC: index.sdoc .................................... 0.00s
Reading SDOC: SC_SafetyManual.sdoc .......................... 0.00s
Reading SDOC: c_library.sdoc ................................ 0.01s
Reading SDOC: interrupts.sdoc ............................... 0.00s
Reading SDOC: logging.sdoc .................................. 0.00s
Reading SDOC: memory_objects.sdoc ........................... 0.00s
Reading SDOC: memory_protection.sdoc ........................ 0.00s
Reading SDOC: mutex.sdoc .................................... 0.00s
Reading SDOC: power_management.sdoc ......................... 0.00s
Reading SDOC: semaphore.sdoc ................................ 0.00s
Reading SDOC: software_requirements.sgra .................... 0.00s
Reading SDOC: system_initialization.sdoc .................... 0.00s
Reading SDOC: thread_communication.sdoc ..................... 0.00s
Reading SDOC: thread_scheduling.sdoc ........................ 0.00s
Reading SDOC: timers.sdoc ................................... 0.01s
Reading SDOC: tracing.sdoc .................................. 0.00s
Reading SDOC: threads.sdoc .................................. 0.00s
Reading SDOC: index.sdoc .................................... 0.00s
Reading SDOC: system_requirements.sgra ...................... 0.00s
Reading SDOC: SC_FunctionalSafetyManagementPlan.sdoc ........ 0.00s
Reading SDOC: SC_FunctionalSafetyVerificationPlan.sdoc ...... 0.00s
Reading SDOC: SC_FunctionalSafety_Certification_Strategy.sdoc . 0.01s
Reading SDOC: SC_IEC61508_techniques_measures.sdoc .......... 0.00s
Reading SDOC: SC_ToolEvaluation_Qualification.sdoc .......... 0.00s
Reading SDOC: SWG_ChangeManagementPlan.sdoc ................. 0.00s
Reading SDOC: SWG_ConfigurationManagementPlan.sdoc .......... 0.00s
Reading SDOC: SWG_FunctionalSafetyManagement_Guideline.sdoc . 0.00s
Reading SDOC: SWG_Glossary.sdoc ............................. 0.01s
Reading SDOC: SWG_ReqMangement_Guideline.sdoc ............... 0.01s
Reading SDOC: SWG_Requirements Creation Guideline and Checklist.sdoc . 0.00s
Reading SDOC: SWG_ToolChainUsage_Guideline.sdoc ............. 0.00s
Step 'Find and read SDoc files' took: 5.58 sec.
Step 'Build traceability graph' start.
error: TraceabilityIndex: the document MID is not unique: 1de8e492b37445f79880996ac14b9e5d. All machine identifiers (MID) must be unique values.
```

**Comment by stanislaw** [comment]
Thanks a lot for reporting this! Most of the error messages are normally reporting the problematic documents/nodes correctly but this particular message needs an improvement.

We will work out a fix within a few days.

**Comment by stanislaw** [comment]
As a quick fix, for now, you could use `grep -r "1de8e492b37445f79880996ac14b9e5d" ./` to find a problematic file in a current directory.

**Comment by stanislaw** [comment]
I have just released a nightly version which is available using `pip install strictdoc --pre`. The error messages should be improved now. Let me know if it works better now.

The main release will be made next week or earlier and will contain many other changes, so I prefer to postpone releasing this specific change right now.

Thanks again for reporting this.
